pkg/nuctl/command/common/renderers.go

```diff
@@ -23,20 +23,21 @@ const (
 	OutputFormatYAML = "yaml"
 )
 
-func RenderFunctions(logger logger.Logger,
+func RenderFunctions(ctx context.Context,
+	logger logger.Logger,
 	functions []platform.Function,
 	format string,
 	writer io.Writer,
 	renderCallback func(functions []platform.Function, renderer func(interface{}) error) error) error {
 
-	errGroup, _ := errgroup.WithContext(context.Background(), logger)
+	errGroup, errGroupCtx := errgroup.WithContext(ctx, logger)
```

Comment:
I just realized, nothing waits on `errGroup`. make sure it happens please

Reply:
I also noticed that the errGroup goroutine never returns an error - only logs it if function initialization fails.
I'll add a `Wait()` but it will never fail...